cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -169,6 +169,19 @@ inline compute::Expression UseBoringRefs(const compute::Expression& expr) {
   return compute::Expression{std::move(modified_call)};
 }
 
+Status CheckTable(std::shared_ptr<Buffer>& buf,
+                  const ConversionOptions& conversion_options = {}) {
+  std::shared_ptr<ExtensionIdRegistry> sp_ext_id_reg = MakeExtensionIdRegistry();
+  ExtensionIdRegistry* ext_id_reg = sp_ext_id_reg.get();
+  ExtensionSet ext_set(ext_id_reg);
+  ARROW_ASSIGN_OR_RAISE(auto sink_decls, DeserializePlans(
+                                             *buf, [] { return kNullConsumer; },
+                                             ext_id_reg, &ext_set, conversion_options));
```

Comment:
You can pass `nullptr` for `ext_id_reg` and `ext_set` and get the same behavior I'm pretty sure.

Reply:
Removed.